UnitsNet/GeneratedCode/Quantities/AmplitudeRatio.g.cs

```diff
@@ -187,6 +214,33 @@ public static AmplitudeRatio FromDecibelMillivolts(double decibelmillivolts)
         }
 
         /// <summary>
+        ///     Get AmplitudeRatio from DecibelMillivolts.
+        /// </summary>
+        public static AmplitudeRatio FromDecibelMillivolts(int decibelmillivolts)
+        {
+            return new AmplitudeRatio(decibelmillivolts - 60);
+        }
+
+        /// <summary>
+        ///     Get AmplitudeRatio from DecibelMillivolts.
+        /// </summary>
+        public static AmplitudeRatio FromDecibelMillivolts(long decibelmillivolts)
+        {
+            return new AmplitudeRatio(decibelmillivolts - 60);
+        }
+
+        // Windows Runtime Component does not support decimal type
+#if !WINDOWS_UWP
+        /// <summary>
+        ///     Get AmplitudeRatio from DecibelMillivolts of type decimal.
+        /// </summary>
+        public static AmplitudeRatio FromDecibelMillivolts(decimal decibelmillivolts)
+        {
+            return new AmplitudeRatio(Convert.ToDouble(decibelmillivolts) - 60);
+        }
```

Comment:
I had an idea, but not sure if I prefer it or not myself:
```
public static double From<TNumber>(TNumber val) where TNumber : struct, IConvertible
{
    return Convert.ToDouble(val) - 60;
}
```

You get int, long, float, double, decimal all in one method - but the method signature is probably a bit weird when consuming it and it may also accept struct types that implement `IConvertible`. I just tried `TimeSpan` and it did not accept that type at least, so not sure how much of a problem that is. `TNumber` should hopefully indicate we are looking for a number type and Xmldoc could also be added for `<typeparam`> to describe what we expect here.
Ideally, c# added type constraints for only matching on number primitives and doing arithmetic on generic numbers, but it doesn't have that although I have seen some open source libs for that.

Reply:
I didn't know of the `IConvertible` interface. I like the the current implementation better though, since we already do all the code generation I'm less worried about brevity and it has no side effects/unexpected behaviors.

Is there a particular reason we use `Convert` instead of type casting to double?

Reply:
Hm, not sure. I think it was to be on the safe side, as Convert methods do has some additional magic - such as rounding when converting to integers, whereas casting just floors the int.
I don't know if it makes a practical difference for ToDouble() though, I haven't looked into decimal to double cast vs conversion behavior.

Reply:
Checked the [reference source](https://referencesource.microsoft.com/#mscorlib/system/convert.cs,1596) and it casts internally. That method call will most likely be inlined by the JIT compiler so there should be no practical difference.